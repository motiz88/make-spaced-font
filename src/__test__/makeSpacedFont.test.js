const fs = require("fs");
const tmp = require("tmp-promise");
const { promisify } = require("util");
const copyFile = promisify(fs.copyFile);
const exists = promisify(fs.exists);

const makeSpacedFont = require("../");

const s544b_ttf = __dirname + "/../../fixtures/s544b.ttf";

describe("makeSpacedFont", () => {
  let dir;
  beforeEach(async () => {
    dir = await tmp.dir({ unsafeCleanup: true });
    await copyFile(s544b_ttf, dir.path + "/s544b.ttf");
  });
  afterEach(() => {
    dir.cleanup();
  });
  test("throws if input doesn't exist", async () => {
    expect.assertions(1);
    try {
      await makeSpacedFont({
        inputFile: "SOME_FILE_THAT_DOES_NOT_EXIST",
        letterSpacing: "1em"
      });
    } catch (e) {
      expect(e).toMatchSnapshot();
    }
  });
  test("add 1em to test ttf", async () => {
    const outputFile = dir.path + "/s544b.out.ttf";
    await makeSpacedFont({
      inputFile: dir.path + "/s544b.ttf",
      outputFile,
      letterSpacing: "1em"
    });
    expect(await exists(outputFile)).toBe(true);
  });
  test("add -0.1em to test ttf", async () => {
    const outputFile = dir.path + "/s544b.out.ttf";
    await makeSpacedFont({
      inputFile: dir.path + "/s544b.ttf",
      outputFile,
      letterSpacing: "-0.1em"
    });
    expect(await exists(outputFile)).toBe(true);
  });
  test("add 1px with 12px base to test ttf", async () => {
    const outputFile = dir.path + "/s544b.out.ttf";
    await makeSpacedFont({
      inputFile: dir.path + "/s544b.ttf",
      outputFile,
      letterSpacing: "1px",
      baseFontSize: "12px"
    });
    expect(await exists(outputFile)).toBe(true);
  });
});
