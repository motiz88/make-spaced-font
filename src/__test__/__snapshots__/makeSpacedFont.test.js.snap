// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`makeSpacedFont throws if input doesn't exist 1`] = `[Error: File not found: SOME_FILE_THAT_DOES_NOT_EXIST]`;
